From 9bea4d0ebdaeef8745e4932e6c4891942d5dc895 Mon Sep 17 00:00:00 2001
From: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
Date: Fri, 4 Dec 2020 14:09:49 +0530
Subject: [PATCH] Fix to make BT Chip boot with latest firmware every boot

This fix forces BT chip in operational mode to Bootloader
mode, so that the latest firmware is downloaded.

Tracked-On: OAM-93634
Signed-off-by: Aiswarya Cyriac <aiswarya.cyriac@intel.com>
---
 drivers/bluetooth/btusb.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/bluetooth/btusb.c b/drivers/bluetooth/btusb.c
index fd809db72e47..626a9a25ca3d 100644
--- a/drivers/bluetooth/btusb.c
+++ b/drivers/bluetooth/btusb.c
@@ -2387,7 +2387,15 @@ static int btusb_setup_intel_new(struct hci_dev *hdev)
 		btintel_reset_to_bootloader(hdev);
 		return err;
 	}
+	/* Below fix is to make sure each time device boots with latest
+	 * firmware.
+	 */
+	if (ver.fw_variant != 0x06) {
 
+		bt_dev_info(hdev,"Bluetooth: Entered bootloader mode");
+		btintel_reset_to_bootloader(hdev);
+		return 0;
+	}
 	/* The hardware platform number has a fixed value of 0x37 and
 	 * for now only accept this single value.
 	 */
-- 
2.17.1

